server/src/internalClusterTest/java/org/elasticsearch/indices/IndicesLifecycleListenerIT.java

```diff
@@ -143,6 +144,21 @@ public void beforeIndexCreated(Index index, Settings indexSettings) {
         }
         updateIndexSettings(Settings.builder().put(INDEX_ROUTING_EXCLUDE_GROUP_PREFIX + "._name", node1), "index1");
         ensureGreen("index1");
+
+        var state = clusterAdmin().prepareState().get().getState();
```

Comment:
was this test also affected? Or are you adding the same functionality to be prudent?

Reply:
This test case perform the similar action so I added an additional assertion to check the end state